resultbrowser/src/js/jsonview.js

```diff
@@ -18,4 +18,403 @@
+                const parsedData = JSON.parse(jsonData);
+
+                tree = createTree(parsedData);
+
+                tree.render(targetElement);
+                tree.collapseAll();
+                tree.expand();
+            }
+            catch (error) {
+                targetElement.textContent = error;
+            }
+        },
+
+        /**
+         * Collapses all JSON nodes recursively.
+         */
+        collapseAll: function () {
+            tree && tree.collapseAll();
+        },
+
+        /**
+         * Expands all JSON nodes recursively.
+         */
+        expandAll: function () {
+            tree && tree.expandAll();
+        },
+
+        /**
+         * The CSS selector for the DOM element which holds the number of matches found.
+         */
+        matchesElementSelector: '#matches',
+
+        /**
+         * Searches the JSON tree for entries matching the search phrase and marks any match.
+         *
+         * @param {string} searchPhrase - the text to search for
+         * @param {boolean} ignoreCase - wehther the search is case insensitive
+         * @param {boolean} filter - whether non-matching lines should be filtered out
+         */
+        search: function (searchPhrase, ignoreCase, filter) {
+            if (tree) {
+                searchPhrase = searchPhrase || '';
+                searchState = { matches: [], currentMatch: undefined, ignoreCase: ignoreCase, filter: filter, };
+
+                tree.search(searchPhrase, searchState);
+                this.highlightNextMatch(true);
+
+                // update number of matches found
+                document.querySelector(this.matchesElementSelector).textContent = searchState.matches.length;
+            }
+        },
+
+        /**
+         * Highlights the next occurrence of the search phrase in the JSON tree.
+         *
+         * @param {boolean} forward - whether to go forward or backward
+         */
+        highlightNextMatch: function (forward) {
+            if (searchState && searchState.matches.length > 0) {
+                if (searchState.currentMatch === undefined) {
+                    searchState.currentMatch = 0;
+                }
+                else {
+                    searchState.matches[searchState.currentMatch].classList.remove('json-match-current');
+
+                    if (forward) {
+                        searchState.currentMatch = (searchState.currentMatch == searchState.matches.length - 1) ? 0 : searchState.currentMatch + 1;
+                    }
+                    else {
+                        searchState.currentMatch = (searchState.currentMatch == 0) ? searchState.matches.length - 1 : searchState.currentMatch - 1;
+                    }
+                }
+
+                searchState.matches[searchState.currentMatch].scrollIntoView(!forward);
+                searchState.matches[searchState.currentMatch].classList.add('json-match-current');
+            }
+        },
+    }
+
+})();
```

Comment:
Please add a final new line character at end-of-file.

Reply:
Done.